src/solve.jl

```diff
@@ -2,8 +2,10 @@ function SciMLBase.solve(prob::NonlinearProblem,
                          alg::AbstractNonlinearSolveAlgorithm, args...;
                          kwargs...)
   solver = init(prob, alg, args...; kwargs...)
-  sol = solve!(solver)
-  return sol
+  sol, resid = solve!(solver)
+  if typeof(sol) <: NewtonSolution
```

Comment:
Why is this branching needed? It makes the solution type unstable and break non-Newton solvers. We can just assume that `sol.u` exists.

Reply:
There are two types of solver, Newton solver and Bracketing methods, which outputs `sol.left` and `sol.right`. That's why I wanted to ask how to pass bracketing methods to `build_solution`?